### ISSUE TITLE
End of Turn/Negative Cargo Bug - Crashing the Game

### ISSUE THREAD

**Issue by VihanJ** [We-the-People-civ4col-mod/Mod#620]
The error, shown below. I only got this after doing some worldbuilder shenanigans.
![Screenshot (37)](https://user-images.githubusercontent.com/55373538/189314928-1e4b5117-62f6-4b48-a608-261b8a5bf0ff.png)

Reproducing the error:
Load the save into Worldbuilder and save as a worldbuilder save. Reload the worldbuilder save, and load in as the Dutch player. 
The error should pop-up immediately (there are many error windows, with different error codes)

After closing all error windows, all settlements have 0 cargo. The settlement below has negative cargo.  Immediately obvious are how the tools and blades have values of -50, the printing press, for some reason, costs 70 blades and 50 tools, and the colonists outside of the town are rearranged. 
![Screenshot (38)](https://user-images.githubusercontent.com/55373538/189314933-38b6ec67-f00c-4950-b23c-69614f707ca0.png)

Here is what the settlement looked like before worldbuilder shenanigans.
![Screenshot (39)](https://user-images.githubusercontent.com/55373538/189314938-25c69b95-27c8-4bb4-a604-90f121dfd94d.png)

**Comment by VihanJ** [comment]
The save file:
[Meme AD-1598-JanuaryBackup.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/9534109/Meme.AD-1598-JanuaryBackup.zip)

If any other things are needed, then please let me know.

**Comment by VihanJ** [comment]
Additionally, I somehow got an autosave file after loading the worldbuilder. There are no interactions with the natives, but there is no more negative cargo either. The positions of all units are preserved

Here is the autosave
[AutoSave_Initial_AD-1598-January.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/9534184/AutoSave_Initial_AD-1598-January.zip)

**Comment by VihanJ** [comment]
> Additionally, I somehow got an autosave file after loading the worldbuilder. There are no interactions with the natives, but there is no more negative cargo either (and all settlements have 0 cargo). The positions of all units are preserved
> 
> Here is the autosave [AutoSave_Initial_AD-1598-January.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/9534184/AutoSave_Initial_AD-1598-January.zip)

Starting from the autosave allows me to end the turn. After the turn, all events from the beginning of the game are repeated, including the event for the basecamp upon creating your first settlement (except this event applied to my third settlement)

**Comment by VihanJ** [comment]
> The error, shown below. I only got this after doing some worldbuilder shenanigans.

Using the same worldbuilder file to repeat this process with other players resulted in me seeing that the French and Russian players also had negative cargo, but with negative values only for guns in at least 2 of their settlements (and all other cargo was 0)

Additionally, using the same worldbuilder file to repeat this process for 2 or so attempts would eventually fail, with the game crashing after selecting a player character and loading into the game, with the game stuck on the loading screen. A new auto save file would still be generated, but attempting to load in using this file would also crash the game.

**Comment by VihanJ** [comment]
> The error, shown below. I only got this after doing some worldbuilder shenanigans.

I can repeatedly load the worldbuilder file using the swedish and russian players, both of whom have negative values for guns. However, I cannot load them for French or Spanish players, always resulting in a crash. Loading the resulting autosave files created after loading the world using French or Spanish also crashes the game.

**Comment by VihanJ** [comment]
I also did a genocide of all players using worldbuilder to find out if a unit/settlement is causing the issue. I found that the Russians had a ghost caravel:
![Screenshot (45)](https://user-images.githubusercontent.com/55373538/189331583-4011b1ff-a8fc-40b7-8d62-ac8a3e871ac6.png)
[Ghost Russian Caravel.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/9534630/Ghost.Russian.Caravel.zip)

Everytime I removed this unit from the worldbuilder and exited, the unit would show up again. I could eliminate every colonizer and native except for Russia because of this ghost unit.
